tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SdkCodeTool.cs

```diff
@@ -0,0 +1,473 @@
+                }
+
+                var scriptPath = element.GetString();
+                if (string.IsNullOrEmpty(scriptPath))
+                {
+                    throw new InvalidOperationException($"Script path is empty at JSON path '{jsonPath}' in configuration file {configFilePath}.");
+                }
+
+                return scriptPath;
+            }
+            catch (JsonException ex)
+            {
+                throw new InvalidOperationException($"Error parsing JSON configuration: {ex.Message}", ex);
+            }
+        }
+
+        // Try to get a JSON element by its path
+        private (bool found, JsonElement element) TryGetJsonElementByPath(JsonElement root, string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return (false, default);
+            }
+
+            var pathParts = path.Split('/', StringSplitOptions.RemoveEmptyEntries);
+            JsonElement current = root;
+
+            foreach (var part in pathParts)
+            {
+                if (!current.TryGetProperty(part, out current))
+                {
+                    return (false, default);
+                }
+            }
+
+            return (true, current);
+        }
+
+        // Validate commitSha: must be a 40-character hex string
+        private bool IsValidSha(string sha)
+        {
+            if (!string.IsNullOrEmpty(sha))
+            {
+                var match = Regex.Match(sha, @"^[a-f0-9]{40}$", RegexOptions.IgnoreCase);
+                return match.Success;
+            }
+
+            return false;
+        }
+
+        // Validate remote GitHub URL with commit SHA
+        private bool IsValidRemoteGitHubUrlWithCommit(string tspConfigPath)
+        {
+            // Must contain /blob/ pattern
+            if (!tspConfigPath.Contains("/blob/"))
+            {
+                return false;
+            }
+
+            // Extract the part after /blob/ and before the next /
+            var blobIndex = tspConfigPath.IndexOf("/blob/", StringComparison.OrdinalIgnoreCase);
+            if (blobIndex == -1)
+            {
+                return false;
+            }
+
+            var afterBlob = tspConfigPath.Substring(blobIndex + 6);
+            var nextSlashIndex = afterBlob.IndexOf('/');
+            if (nextSlashIndex == -1)
+            {
+                return false;
+            }
+
+            var commitOrBranch = afterBlob.Substring(0, nextSlashIndex);
+            
+            // Validate that it's a 40-character commit SHA, not a branch name
+            return IsValidSha(commitOrBranch);
+        }
+
+        // Helper method to create failure responses along with setting the failure state
+        private DefaultCommandResponse CreateFailureResponse(string message)
+        {
+            SetFailure();
+            return new DefaultCommandResponse
+            {
+                Result = "failed",
```

Comment:
Result can be ommitted we won't even return it in serialization when errors are set.

Reply:
Removed it.